### PR TITLE
[programmatic-payments] Stored payment methods

### PR DESCRIPTION
**Pull request by korycins**
It adds:
- section to payments: `stored-payment-methods`
- description of the list-stored-payment-methods webhook

More details can be found [here](https://github.com/saleor/saleor/issues/13384)

### CODE REVIEW

File: docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+# Introduction
+
+:::caution
+This feature is in the **Feature Preview** stage, which means that it is available for experimentation and
+feedback. However, it is still undergoing development and is subject to modifications.
+:::
+
+A stored payment method is a payment method that can be used later by customers without providing all payment details.
+Saleor has synchronous webhooks that allow delegating the stored payment method's management to the payment app.
```
Comment (witoszekdev) [inline docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx:13]: > without providing all payment details

I would add an example:

> without providing all payment details (e.g. 1-click checkout)

File: docs/developer/payments.mdx [hunk 2]
```diff
@@ -952,6 +952,52 @@ and its amount is added to `transaction.chargedAmount`. Finally, the amount of t
 
 ---
 
+### Stored payment methods
+
+A stored payment method is a payment method that can be used later by customers without providing all payment details.
+Saleor uses synchronous webhooks to notify the payment app about actions (such as list-stored-payment-method) requested by the customer.
```
Comment (witoszekdev) [inline docs/developer/payments.mdx:958]: I would add a link to `list-stored-payment-method`: https://saleor-docs-1juwt4dfh-saleorcommerce.vercel.app/docs/3.x/developer/extending/webhooks/synchronous-events/stored-payment-method#list-stored-payment-methods

**Comment by patrys** [comment]
Can we regenerate docs on `main` and then redo this PR? It rewrites the entire API reference 😢

**Comment by 2can** [comment]
@patrys @korycins docs regenerated here https://github.com/saleor/saleor-docs/pull/858

**Comment by 2can** [comment]
@korycins https://github.com/saleor/saleor-docs/pull/858 merged, please rebase or re-generate

**Comment by korycins** [comment]
@2can  @patrys  branch rebased. I've re-generated the api-reference based on the schema from my feature branch. Still, there are a lot of modified fields. We should agree to always generate the schema from main saleor branch or figure out a different approach.

**Comment by patrys** [comment]
I don't know how to review this 😢

File: docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      // "creditCardInfo": [Optional] credit card information if the payment method is a credit card.
+      "creditCardInfo": {
+        "brand": "<credit card bran>",
+        "lastDigits": "<last digits>",
+        "expMonth": "<expiration month>",
+        "expYear": "<expiration year>",
+        "firstDigits": "<[Optional] first digits>"
+      },
+      "name": "<[Optional] name of the payment method. Example: last 4 digits of credit card, obfuscated email>",
+      "data": "<[Optional] JSON data tha will be returned to storefront>"
```
Comment (mmiszy) [inline docs/developer/extending/webhooks/synchronous-events/stored-payment-method.mdx:108]:
```suggestion
      "supportedPaymentFlows": "<list of supported flows that can be performed with this payment method>",
      "type": "<type of the payment method. For example: Credit Card>",
      // "creditCardInfo": [Optional] credit card information if the payment method is a credit card.
      "creditCardInfo": {
        "brand": "<credit card brand>",
        "lastDigits": "<last digits>",
        "expMonth": "<expiration month>",
        "expYear": "<expiration year>",
        "firstDigits": "<[Optional] first digits>"
      },
      "name": "<[Optional] name of the payment method. For example: last 4 digits of credit card, obfuscated email>",
      "data": "<[Optional] JSON data that will be returned to storefront>"
```

**Review by mmiszy — CHANGES_REQUESTED** [review]

**Review by mmiszy — APPROVED** [review]

**Review by 2can — APPROVED** [review]

**Review by peelar — APPROVED** [review]

**Pull request merged**